sw/host/opentitantool/src/command/lc.rs

```diff
@@ -133,6 +138,10 @@ impl CommandDispatch for LcRegRead {
     ) -> Result<Option<Box<dyn Annotate>>> {
         // Set the TAP straps for the lifecycle controller.
         transport.pin_strapping("PINMUX_TAP_LC")?.apply()?;
+        // Apply bootstrap pin to be able to connect to JTAG when ROM execution is
+        // enabled.
+        transport.pin_strapping("ROM_BOOTSTRAP")?.apply()?;
+        transport.reset_target(self.reset_delay, true)?;
```

Comment:
Personally, I think it would be best if this command remained focus on its particular job. If we do a reset here, it's rather cumbersome to inspect the device: We'd get a reset every time, where we might just want to see the current value.

Instead, perhaps we should consider making use of `--exec` in docs to provide the initial setup with a sequence. Or we could have an option to perform initialization for this command, I suppose, but I'd suggest that it shouldn't be the default behavior. What do you think?

Reply:
If the docs were to mention a separate initialization sequence, it would look something like this:
```sh
./bazel-out/k8-fastbuild/bin/sw/host/opentitantool/opentitantool \
    --interface hyper310 \
    --exec "transport init" \
    --exec "gpio apply PINMUX_TAP_LC" \
    --exec "gpio apply ROM_BOOTSTRAP" \
    --exec "gpio write RESET false" \
    --exec "gpio write RESET true" \
    no-op
```

Reply:
There are two scenarios where this command could be used:
1. in test_unlocked / RMA, where the TAP straps are continuously sampled, or 
2. in mission mode (DEV / PROD*) where they are only sampled once at boot time (and the latter of which closes off the RISCV TAP.

If we are scenario 1, we don't need to reset to activate the new TAP. If we are in scenario 2, we would have needed to apply the straps _before_ the reset. However, resetting could destroy the state we are trying to read. So, I agree we probably don't want to reset for this particular command. However setting the bootstrap pins won't hurt, since if we are in either scenarios 1 or 2, if we are in a reset loop due to an empty flash, we could break the reset loop by setting the bootstrap pins.

However, the other LC commands I believe we should reset to apply the TAP selection correctly, since those commands read persistent state / perform persistent state-altering actions (like an LC transition).

Reply:
I agree that the transition commands are clearly processes that would take over any running session and reset. :)

Setting the bootstrap pins for regread _can_ be harmful, though. Remember that those pins are straps only, and they can be reused for other purposes after boot--There are some tests that use them to signal advancing to the next phase.